crates/api/src/local_user/save_settings.rs

```diff
@@ -27,150 +26,123 @@ use lemmy_utils::{
+      let url =
+        build_totp_2fa(&site_view.site.name, &local_user_view.person.name, &secret)?.get_url();
+      (Some(Some(secret)), Some(Some(url)))
     } else {
-      (None, None)
-    };
-
-    let local_user_form = LocalUserUpdateForm::builder()
-      .email(email)
-      .show_avatars(data.show_avatars)
-      .show_read_posts(data.show_read_posts)
-      .show_new_post_notifs(data.show_new_post_notifs)
-      .send_notifications_to_email(data.send_notifications_to_email)
-      .show_nsfw(data.show_nsfw)
-      .blur_nsfw(data.blur_nsfw)
-      .auto_expand(data.auto_expand)
-      .show_bot_accounts(data.show_bot_accounts)
-      .show_scores(data.show_scores)
-      .default_sort_type(default_sort_type)
-      .default_listing_type(default_listing_type)
-      .theme(theme)
-      .interface_language(data.interface_language.clone())
-      .totp_2fa_secret(totp_2fa_secret)
-      .totp_2fa_url(totp_2fa_url)
-      .open_links_in_new_tab(data.open_links_in_new_tab)
-      .infinite_scroll_enabled(data.infinite_scroll_enabled)
-      .build();
-
-    let local_user_res =
-      LocalUser::update(&mut context.pool(), local_user_id, &local_user_form).await;
-    let updated_local_user = match local_user_res {
-      Ok(u) => u,
-      Err(e) => {
-        let err_type = if e.to_string()
-          == "duplicate key value violates unique constraint \"local_user_email_key\""
-        {
-          LemmyErrorType::EmailAlreadyExists
-        } else {
-          LemmyErrorType::UserAlreadyExists
-        };
-
-        return Err(e).with_lemmy_type(err_type);
-      }
-    };
-
-    // Return the jwt
-    Ok(LoginResponse {
-      jwt: Some(
-        Claims::jwt(
-          updated_local_user.id.0,
-          &context.secret().jwt_secret,
-          &context.settings().hostname,
-        )?
-        .into(),
-      ),
-      verify_email_sent: false,
-      registration_created: false,
-    })
-  }
+      (Some(None), Some(None))
+    }
+  } else {
+    (None, None)
+  };
+
+  let local_user_form = LocalUserUpdateForm::builder()
+    .email(email)
+    .show_avatars(data.show_avatars)
+    .show_read_posts(data.show_read_posts)
+    .show_new_post_notifs(data.show_new_post_notifs)
+    .send_notifications_to_email(data.send_notifications_to_email)
+    .show_nsfw(data.show_nsfw)
+    .blur_nsfw(data.blur_nsfw)
+    .auto_expand(data.auto_expand)
+    .show_bot_accounts(data.show_bot_accounts)
+    .show_scores(data.show_scores)
+    .default_sort_type(default_sort_type)
+    .default_listing_type(default_listing_type)
+    .theme(theme)
+    .interface_language(data.interface_language.clone())
+    .totp_2fa_secret(totp_2fa_secret)
+    .totp_2fa_url(totp_2fa_url)
+    .open_links_in_new_tab(data.open_links_in_new_tab)
+    .infinite_scroll_enabled(data.infinite_scroll_enabled)
+    .build();
+
+  LocalUser::update(&mut context.pool(), local_user_id, &local_user_form).await?;
+
+  Ok(Json(()))
```

Comment:
Did some cleanup in this file. It doesnt try to parse postgres error strings anymore, and doesnt generate a new jwt token unnecessarily.